src/test/rpc/Book_test.cpp

```diff
@@ -800,6 +800,460 @@ class Book_test : public beast::unit_test::suite
             books)[jss::status] == "success");
     }
 
+    void testTrackOffers(void)
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account gw {"gw"};
+        auto wsc = makeWSClient(env.app().config());
+        env.fund(XRP(20000), "alice", "bob", gw);
+        env.close();
+        auto USD = gw["USD"];
+
+        Json::Value books;
+        {
+            books[jss::books] = Json::arrayValue;
+            {
+                auto &j = books[jss::books].append(Json::objectValue);
+                j[jss::snapshot] = true;
+                j[jss::taker_gets][jss::currency] = "XRP";
+                j[jss::taker_pays][jss::currency] = "USD";
+                j[jss::taker_pays][jss::issuer] = gw.human();
+            }
+
+            auto jv = wsc->invoke("subscribe", books);
+            BEAST_EXPECT(jv[jss::status] == "success");
+            BEAST_EXPECT(jv[jss::result].isMember(jss::offers) &&
+                jv[jss::result][jss::offers].size() == 0);
+            BEAST_EXPECT(! jv[jss::result].isMember(jss::asks));
+            BEAST_EXPECT(! jv[jss::result].isMember(jss::bids));
+        }
+
+        env(rate(gw, 1.1));
+        env.close();
+        env.trust( USD(1000), "alice" );
+        env.trust( USD(1000), "bob" );
+        env(pay(gw, "alice", USD(100)));
+        env(pay(gw, "bob",   USD(50)));
+        env(offer("alice", drops(4000), USD(10)));
+        env.close();
+
+        Json::Value jvParams;
+        jvParams[jss::taker] = env.master.human();
+        jvParams[jss::taker_pays][jss::currency] = "XRP";
+        jvParams[jss::ledger_index] = "validated";
+        jvParams[jss::taker_gets][jss::currency] = "USD";
+        jvParams[jss::taker_gets][jss::issuer] = gw.human();
+        auto const jrr =
+            env.rpc("json", "book_offers", to_string(jvParams)) [jss::result];
+        env.close();
+        BEAST_EXPECT(jrr[jss::offers].isArray());
+        BEAST_EXPECT(jrr[jss::offers].size() == 1);
+        auto const jrOffer = jrr[jss::offers][0u];
+        BEAST_EXPECT(jrOffer[sfAccount.fieldName] == Account("alice").human());
+        BEAST_EXPECT(jrOffer[sfBookDirectory.fieldName] ==
+            "1C5C34DB7DBE43E1EA72EE080416E88A87C18B2AD29BD8C4570E35FA931A0000");
```

Comment:
Not sure if its worth doing but you could grab the value from the book. It would look something like this.
```    auto uBookBase = getBookBase({XRP, USD});
    auto uBookEnd = getQualityNext(uBookBase);
    auto view = env.closed();
    auto sleOfferDir = view->read(keylet::page(
        *view->succ(uBookBase, uBookEnd)));
    uint256 offerIndex;
    unsigned int bookEntry;
    cdirFirst(*view, sleOfferDir->key(), sleOfferDir,
        bookEntry, offerIndex, {});
    auto sleOffer = view->read(keylet::offer(offerIndex));
    auto dir = to_string(sleOffer->getFieldH256(sfBookDirectory));
```